### PR TITLE
PD-2492: SearchInput v0.9.0

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

This does not implement the typeahead search yet - waiting for engineering specs and prioritizing AWS deployment of the website before it. This is a simple `<input type="text">` with the appropriate styles.

🎟 _Jira ticket:_ [PD-2492](https://jira.mongodb.org/browse/PD-2492)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [x] I have added my new package to the global tsconfig
- [x] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/search-input/tsconfig.json
Comment (shaneeza) [inline packages/search-input/tsconfig.json:12]: You're missing a few packages

**Review by shaneeza — COMMENTED** [review]
Whats the reason for using `input` instead of using `TextInput`?

File: packages/search-input/tsconfig.json
Comment (spark33) [reply packages/search-input/tsconfig.json:12]: Thanks for catching it! I always forget this has to be manual.

**Comment by spark33** [comment]
> Whats the reason for using `input` instead of using `TextInput`?

`TextInput` comes with other functionalities like the `label` and `description` props. There will also be style changes in each component. that are independent of each other. For the styles and other shared characteristics, I'm a fan of implementing something like `BaseTextInput` or something that all text-type `<input>` components can use. Until we have something like that, I think it would be best to keep them all independent of one another.

File: packages/search-input/tsconfig.json
Comment (shaneeza) [reply packages/search-input/tsconfig.json:12]: actually i think Adam added a scrip to add these

Comment (shaneeza) [reply packages/search-input/tsconfig.json:12]: `check:deps`

**Review by shaneeza — COMMENTED** [review]
Are we adding this to the `website`?

**Comment by spark33** [comment]
> Are we adding this to the `website`?

Yes! I was thinking we would just add it to the new website after we published it since we're getting pretty close to launch on that front. This is also a v0.9 that doesn't have all the props we want (i.e. typeahead, loading state, etc), so I wasn't too pressed on getting it on the website for now.

**Comment by spark33** [comment]
> Looks good, I did notice that the icon is really tiny when the size is `xsmall`, from the design it looks like its always the same size for each size variant.
> 
> Screen Shot 2022-10-11 at 4 34 09 PM

Oh you're right - I don't know if that's changed in the meanwhile, or I was tripping. I'll change that!

File: packages/search-input/tsconfig.json
Comment (spark33) [reply packages/search-input/tsconfig.json:19]: Hm that script didn't work for me. I updated the file manually anyway!

File: .changeset/two-years-pretend.md
Comment (TheSonOfThomp) [inline .changeset/two-years-pretend.md:6]: Maybe break these out into separate files? So we can have different messages. The patch with this comment, and a second one that releases the search-input

File: packages/search-input/package.json
Comment (TheSonOfThomp) [inline packages/search-input/package.json:39]: Should bump this to `v3.0.0`

File: packages/search-input/src/SearchInput.spec.tsx
Comment (TheSonOfThomp) [inline packages/search-input/src/SearchInput.spec.tsx:19]: Should probably replicate all the `text-input` tests

File: packages/search-input/src/types.ts
Comment (TheSonOfThomp) [inline packages/search-input/src/types.ts:21]: SearchInput should accept `aria-label` as well, and not require `aria-labelledby` in that case

File: packages/text-input/src/TextInput.tsx
Comment (TheSonOfThomp) [inline packages/text-input/src/TextInput.tsx:123]: Should consider using `lib/consoleOnce`, otherwise this will warn on every render

File: scripts/package-builder.js
Comment (TheSonOfThomp) [inline scripts/package-builder.js:150]: 🔥

File: packages/search-input/src/SearchInput.spec.tsx
Comment (spark33) [reply packages/search-input/src/SearchInput.spec.tsx:19]: Some tests aren't relevant for SearchInput (i.e. label, description, baseFontSize, etc.) so there won't be full parity, but I went back and added some more!

File: packages/search-input/src/types.ts
Comment (spark33) [reply packages/search-input/src/types.ts:21]: Actually, I just saw your PR for the TextInput component. What was the discussion there? It would be nice to get on the same page on it before moving forward.

**Comment by sooachung** [comment]
- The disabled state should not have a hover state.
Screen Shot 2022-11-08 at 11 54 12 AM

- The color of the icon in dark mode seems lighter than it should be. Maybe the box outline as well, can you double-check if the colors are right? The icon should be gray light 1 and the outline should be the gray base.
Screen Shot 2022-11-08 at 12 03 02 PM

**Review by sooachung — APPROVED** [review]

**Comment by spark33** [comment]
@TheSonOfThomp This PR feels relevant to PR #1490, so I'll wait for your review before merging this.

File: packages/search-input/src/SearchInput.spec.tsx
Comment (TheSonOfThomp) [inline packages/search-input/src/SearchInput.spec.tsx:60]: I think we should prefer using `userEvent.type` to more closely simulate real user behavior

Comment (TheSonOfThomp) [inline packages/search-input/src/SearchInput.spec.tsx:60]: I think we should prefer using `userEvent.type` to more closely simulate real user behavior

**Review by TheSonOfThomp — APPROVED** [review]

File: packages/search-input/src/types.ts
Comment (brookescarlett) [reply packages/search-input/src/types.ts:19]: Yes definitely valid to accept an `aria-labelledby` prop that relates the label and input field.

**Comment by spark33** [comment]
> > @TheSonOfThomp This PR feels relevant to PR #1490, so I'll wait for your review before merging this.
> 
> As long as there are no conflicts, they can go in together

Cool! Thanks. Merging 👍

**Pull request merged**
